### PR TITLE
Refactored the bot module to be more modular.

### PR DESCRIPTION
**Pull request by Renairisu**

### CODE REVIEW

File: src/bots/chess_bot.rs
Comment (DoormatIka) [inline src/bots/chess_bot.rs:1]: rename this to basic bot (nit) :)

File: src/move_gen.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+/**
+ * Returns the pieces but colored.
+ */
+
+/**
+ * h8Q = promotions
+ * Qh6 = more specific movement (Queen goes to h6)
+ * Qh6xh8 = capturing
+ */
+pub fn get_user_move(board: &Board) -> Result<ChessMove, chess::Error> {
```
Comment (DoormatIka) [inline src/move_gen.rs]: this should be in a separate file

**Review by DoormatIka — CHANGES_REQUESTED** [review]
<3

File: src/move_gen.rs (hunk 1 above)
Comment (Renairisu) [reply src/move_gen.rs]: No

File: src/moves/user_move.rs
Comment (DoormatIka) [inline src/moves/user_move.rs:1]: :stare:

**Review by DoormatIka — APPROVED** [review]

**Pull request merged**